clippy_lints/src/array_indexing.rs

```diff
@@ -1,10 +1,10 @@
 use rustc::lint::*;
 use rustc::middle::const_val::ConstVal;
-use rustc::ty::TyArray;
+use rustc::ty;
 use rustc_const_eval::EvalHint::ExprTypeChecked;
 use rustc_const_eval::eval_const_expr_partial;
 use rustc_const_math::ConstInt;
-use rustc::hir::*;
+use rustc::hir;
```

Comment:
@Manishearth, @llogiq, @oli-obk 
`use rustc::hir;`, `use syntax::ast;` and `use rustc::ty;` make updates like this much easier (ie. sed-able) than `use rustc::hir::*` or `use rustc::ty::TyArray;`. (here there now happens to be something named `TyArray` in both `ty` and `hir`).
Could we decide to prefer that style in the future?